legacy/reqcounter/reqcounter_test.go

```diff
@@ -124,3 +126,35 @@ func TestRequestCounterIncrement(t *testing.T) {
 	// Ensure the request counter was incremented.
 	require.EqualValues(t, &expected, &requestCounter, "The request counter failed to increment its request field.")
 }
+
+func TestCycle(t *testing.T) {
+	// Create a simple request counter expected to log every 3 seconds.
+	requestCounter := NewRequestCounter(82)
+	requestCounter.Interval = time.Second * 3
+	// Collect logging statements.
+	logs := []string{}
+	// Mock logrus.Debug calls.
+	rc.Debug = func(args ...interface{}) {
+		logs = append(logs, fmt.Sprint(args[0]))
+	}
+	// Mock time.
+	fakeTime := tw.FakeTime{
+		Time: defaultTime,
+	}
+	rc.Clock = &fakeTime
+	// Determine the expected logs.
+	expected := []string{
+		"From 2006-01-02 15:04:05 to 2006-01-02 15:04:08 [0 min] there have been 82 requests to GCR.",
+		"From 2006-01-02 15:04:08 to 2006-01-02 15:04:11 [0 min] there have been 0 requests to GCR.",
+		"From 2006-01-02 15:04:11 to 2006-01-02 15:04:14 [0 min] there have been 0 requests to GCR.",
+		"From 2006-01-02 15:04:14 to 2006-01-02 15:04:17 [0 min] there have been 0 requests to GCR.",
+		"From 2006-01-02 15:04:17 to 2006-01-02 15:04:20 [0 min] there have been 0 requests to GCR.",
+	}
+	// Repeatedly run sleep/log cycles.
+	numCycles := 5
```

Comment:
This should be `numCycles := len(expected)`, right?

Reply:
You could even simplify this to generate the expected log values programmatically, given just the number of cycles as the input. This seems simpler than manually writing out the log statements as we're doing now in the `expected` variable. E.g, you could test 100 cycles or 1000 cycles easily if you just generate the `expected` value as part of the test setup before executing the test.

Reply:
@amwat suggested in an earlier PR to explicitly define string literals for expected behavior. It would be nice to generate expected logs, but I think it's more stable & easier to read if we explicitly define them.

Reply:
On second thought, this may be advantageous. We should explore this further in #377

Reply:
Right, so, you could keep this explicit log statement as-is, but for testing multiple test cases of different cycles you could do them programmatically.